test/functional/test_framework/test_node.py

```diff
@@ -353,21 +353,13 @@ def stop_node(self, expected_stderr='', *, wait=0, wait_until_stopped=True):
         for profile_name in tuple(self.perf_subprocesses.keys()):
             self._stop_perf(profile_name)
 
-        # Check that stderr is as expected
-        self.stderr.seek(0)
-        stderr = self.stderr.read().decode('utf-8').strip()
-        if stderr != expected_stderr:
-            raise AssertionError("Unexpected stderr {} != {}".format(stderr, expected_stderr))
-
-        self.stdout.close()
-        self.stderr.close()
-
         del self.p2ps[:]
 
+        assert not expected_stderr or wait_until_stopped  # Must wait to check stderr
```

Comment:
nit, logically equivalent, but likely less ambiguous for the reader's eyes:
```suggestion
        assert (not expected_stderr) or wait_until_stopped  # Must wait to check stderr
```